proposals/0016-c-interface-compiler-library.md

```diff
@@ -50,7 +50,495 @@ older DXC-based solutions to adopt clang as the preferred HLSL compiler.
+    UINT32 *pPartSizeInBytes);
+
+  // Create reflection interface from serialized Dxil container, or 
+  // DXC_PART_REFLECTION_DATA.
+  HRESULT CreateReflection(
+    const DxcBuffer *pData,
+    REFIID iid, void **ppvReflection);
+
+  // Create arguments for IDxcCompiler2::Compile
+  HRESULT BuildArguments(
+    LPCWSTR pSourceName,      // Optional file name for pSource. Used in errors
+                              // and include handlers.
+    LPCWSTR pEntryPoint,      // Entry point name. (-E)
+    LPCWSTR pTargetProfile,   // Shader profile to compile. (-T)
+    LPCWSTR *pArguments,      // Array of pointers to arguments
+    UINT32 argCount,          // Number of arguments
+    const DxcDefine *pDefines,// Array of defines
+    UINT32 defineCount,       // Number of defines
+    IDxcCompilerArgs **ppArgs); // Arguments you can use with Compile() method
+
+  // Takes the shader PDB and returns the hash and the container inside it
+  HRESULT GetPDBContents(
+    IDxcBlob *pPDBBlob, IDxcBlob **ppHash,
+    IDxcBlob **ppContainer);
+};
+```
+
+### Features supported in DirectX shader compiler library (dxcapi.h)
+#### Include Handlers
+Provided to customize the handling of include directives. An implementation of
+the following interface is provided as a hook into compilation. A default
+implementation that reads include files from the filesystem can also be created
+using IDxcUtils::CreateDefaultIncludeHandler.
+
+```c++
+struct IDxcIncludeHandler : public IUnknown {
+  HRESULT STDMETHODCALLTYPE LoadSource(
+    LPCWSTR pFilename,            // candidate filename.
+    IDxcBlob **ppIncludeSource);  // resultant source object for included file,
+                                  // nullptr if not found.
+};
+```
+
+#### Shader Compiler
+IDxcCompiler3 is the most current entrypoint for compiling a shader or 
+disassembling DXIL containers/bitcode.
+
+Provides support for:
+* Compiling a single entry point to the target shader model
+* Compiling a library to a library target (using -R lib_*)
+* Compiling a rootsignature (-T rootsig_*)
+* Preprocessing HLSL source (-P)
+* Disassembling DXIL container or bitcode
+
+```c++
+struct IDxcCompiler3 : public IUnknown {
+  HRESULT Compile(
+    const DxcBuffer *pSource,            // source text to compile
+    LPCWSTR *pArguments,                 // array of pointers to arguments
+    UINT32 argCount,                     // number of arguments
+    IDxcIncludeHandler *pIncludeHandler, // user-provided interface to handle
+                                         // #include directives (optional)
+    REFIID riid, LPVOID *ppResult);      // IDxcResult: status, buffer, and
+                                         // errors
+
+  HRESULT Disassemble(
+    const DxcBuffer *pObject,            // dxil container or bitcode
+    REFIID riid, LPVOID *ppResult);      // IDxcResult: status, disassembly
+                                         // text, and errors
+};
+```
+#### Shader Linker
```

Comment:
nit: although this renders correctly, it looks like github's syntax highlighting gets confused.  I think some vertical space between blocks might help it would a bit.

Reply:
Added vertical whitespace